roles/reproducer/tasks/generate_bm_info.yml

```diff
@@ -121,3 +121,33 @@
     dest: "{{ cifmw_reproducer_controller_basedir }}/parameters/baremetal-info.yml"
     content: "{{ _content | to_nice_yaml }}"
     mode: "0644"
+
+- name: Output ironic_nodes to file
+  when: cifmw_reproducer_ironic_node_name_prefix
+  ansible.builtin.copy:
+    dest: "{{ cifmw_basedir }}/parameters/ironic_nodes.yaml"
+    content: |
+      {% set _ironic_nodes = [] %}
+      {% for node in libvirt_manager_bm_info_data.keys() if node.startswith(cifmw_reproducer_ironic_node_name_prefix) %}
```

Comment:
I'm not too sure how this line will work. `cifmw_reproducer_ironic_node_name_prefix:` usually means/is equivalent to `cifmw_reproducer_ironic_node_name_prefix: null`, and passing null/None to `startswith` is not allowed.

Reply:
If cifmw_reproducer_ironic_node_name_prefix is `null` / `none` the task won't run - the `when` condition will be false and the task is skipped.